src/explorer/model/MavenProjectNode.ts

```diff
@@ -49,14 +50,24 @@ export class MavenProjectNode extends NodeBase {
     public get mavenProject(): MavenProject {
         return this._mavenProject;
     }
+
+    public async isValidPom(): Promise<boolean> {
+        await this.ensureMavenProjectParsed();
+        return !!this._mavenProject;
+    }
+
     private get _hasModules(): boolean {
         return this._mavenProject.modules.length > 0;
     }
 
-    private async _ensureMavenProjectParsed(): Promise<void> {
+    public async ensureMavenProjectParsed(): Promise<void> {
```

Comment:
`ensureMavenProjectParsed` actually doesn't ensure a maven project is successfully parsed, `isValidPom` should always be checked after calling this function. So I think it's better to make this private, and only to expose `isValidPom`.